### PR TITLE
chore(query): make opendal be workspace dep

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -84,6 +84,8 @@ members = [
 # databend maintains:
 openraft = { git = "https://github.com/drmingdrmer/openraft", tag = "v0.7.4-alpha.3" }
 sled = { git = "https://github.com/datafuse-extras/sled", tag = "v0.34.7-datafuse.1", default-features = false }
+opendal = "0.24"
+ordered-float = { version = "3.4.0", default-features = false }
```
Comment (PsiACE) [inline Cargo.toml:88]: We might consider adding features here as well, unless it is otherwise restricted.

Comment (sundy-li) [reply Cargo.toml:88]: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace

Seems the features could be added in children crates.

Comment (sundy-li) [reply Cargo.toml:88]: [features](https://doc.rust-lang.org/cargo/reference/features.html): These are additive with the features declared in the [workspace.dependencies]

**Review by Xuanwo — APPROVED** [review]

**Review by PsiACE — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (PsiACE) [reply Cargo.toml:88]: You're right, my idea was to tell everyone a basic set of features that are used, but given that each crate has its own needs, it seems more logical to specify them in the crate.

**Pull request merged**